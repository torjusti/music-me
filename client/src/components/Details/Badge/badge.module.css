.badge {
  display: inline-flex;
  border: 1px solid #ccc;
  border-radius: 3px;
}

.badge + .badge {
  margin-left: 1rem;
}

.label {
  padding: 0.3rem 0.5rem;
  border-right: 1px solid #ccc;
  background: #efefef;
}

.value {
  padding: 0.3rem 0.5rem;
}
