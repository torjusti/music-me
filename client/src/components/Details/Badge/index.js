import React from 'react';
import styles from './badge.module.css';

const Badge = ({ label, value }) => (
  <div className={styles.badge}>
    <div className={styles.label}>{label}</div>
    <div className={styles.value}>{value}</div>
  </div>
);

export default Badge;
