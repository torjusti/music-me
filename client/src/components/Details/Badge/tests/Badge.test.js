import React from 'react';
import { shallow } from 'enzyme';
import Badge from '../index';
import configureMockStore from 'redux-mock-store'

describe('<Badge />', () => {

  // Create the mock store
  const mockStore = configureMockStore();

  let wrapper, store;

  beforeEach(() => {
    store = mockStore();
    // Shallow render the container passing in the mock store
    wrapper = shallow(
      <Badge store={store} />
    );
  });

  it('should render Badge snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Badge correctly', () => {
    expect(wrapper.length).toEqual(1);
  });
});
