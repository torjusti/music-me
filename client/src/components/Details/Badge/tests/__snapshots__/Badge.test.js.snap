// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render Badge snapshot correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="label"
        />,
        <div
          className="value"
        />,
      ],
      "className": "badge",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "label",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "value",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="label"
          />,
          <div
            className="value"
          />,
        ],
        "className": "badge",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "label",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "value",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
